src/tests/multihost/admultidomain/test_multidomain.py

```diff
@@ -138,12 +138,12 @@ def test_0002_bz2018432(multihost, newhostname, adjoin):
         domain_list_cmd = multihost.client[0].run_command(
             'sssctl domain-list', raiseonerr=False)
         domain_list = domain_list_cmd.stdout_text.split('\n')
-        domain_list.remove("implicit_files")
-        domain_list = domain_list[:-1]
-        multihost_list = multihost.ad
-        multihost_list = multihost_list[:-1]
+        multihost_list = []
+        domain_count = len(domain_list)-1
+        for x in range(domain_count):
+            multihost_list = multihost.ad[x].hostname
```

Comment:
On new L144: multihost_list seems to be filled with hostname of the last AD in the setup here making it a str instead of list.

Later it is used as list of multihost type objects only for domainname attribute to be used.
There seems to be one untrusted ad (last one) according to old L144.
domain_count and multihost_list might not be needed at all.

I think that the right solution is along the lines:
```
for x in multihost.ad[:-1]:
    assert x.domainname in domain_list, f"Domain {x} was not found in the domain list!"
assert multihost.ad[-1].domainname not in domain_list, f"Untrusted domain {multihost.ad[-1].domainname} found!"
```
It should work also to compare domains as sets
```
mh_set =  set([ x.domainname for x in multihost.ad[-1]])
d_set = set(domain_list)
diff = d_set.symmetric_difference(mh_set)
assert not diff, f'Domains do not match: Expected: {", ".join(mh_set)}, Found: {", ".join(d_set)}'
```

Reply:
I rewrote it, removing the items, and simply comparing lists. Please take a look now.